src/UniversalRewardsDistributor.sol

```diff
@@ -98,9 +97,9 @@ contract UniversalRewardsDistributor is IUniversalRewardsDistributor {
         root = pendingRoot.root;
         ipfsHash = pendingRoot.ipfsHash;
 
-        emit EventsLib.RootSet(pendingRoot.root, pendingRoot.ipfsHash);
+        _setPendingRoot(PendingRoot(0, 0, 0));
 
-        delete pendingRoot;
+        emit EventsLib.RootSet(root, ipfsHash);
```

Comment:
Yes I agree with you, imo there is no golden rule for this, we are using the 2 solutions depending on the cases.

Imo it is easier like that, since a pending root can be bypassed by the owner, then tracking the pending root state means having to handle multiple states.

Reply:
> But why don't we expect the integrator to mimick the code's behavior as we do in other smart contracts?

In this specific case
What does it cost? What are the stakes?

I don't see how it's "more complex" for the integrator, compared to the numerous other cases where we only emit a single event whereas we change multiple storage values

For example: `AccrueInterest` in Morpho increases the `feeRecipient`'s supply shares and the market's total supply shares, so we emit `feeRecipient`, the market's `id` and the `supplyShares`
Here, it would be like emitting the new `pendingRoot` = `bytes32(0)` because it known by the implementation ; so I don't argue to emit it, I argue to do nothing and expect the integrator to reset the `pendingRoot` in their system. How is it different?

Reply:
What is the cost of emitting `PendingRootSet ` ?

Reply:
In terms of gas, approximately 1k gas
But my point is not only about the gas cost of emitting it, but also the complexity it adds (the reader needs to check whether the value emitted is correct at all times) and the drastic change in conventions between what we've built up until now and this

Reply:
Good points @Rubilmax, now I understand better what you mean. So now I think we should do what is the most convenient actually, as long as the state can be reconstructed without too much difficulty:
- logging events are a second order problem compared to having working code. One could argue that having clean code is also more important than being very precise and not redundant in how events are logged. If events are factorized in a setter function, we can use that setter so that emitting events is not cluttering the logic of the code
- we can still try to make reconstructing the state not too difficult, for example by adding arguments to the events. The UrdCreated event is a good example of that, where it has additional arguments like the initial root because one can expect that events are tracked after that UrdCreated events, but the first event logging the initial root happens before that

Reply:
And in this case where the value of a storage variable is set to a known value that is not required to be emitted ; should we:
1. emit 2 events, one for each storage variable?
2. emit a single event and expect the integrator to register side effects?
3. emit a single event with the reset value as additional parameter?

I just realized that `lastUpdate` is different because it is ALWAYS set to the block's timestamp (or more generally: to a known value that doesn't need to be logged)
Whereas the pending root can be changed as desired...

Which would motivate 1

Reply:
I think that in this case since we have a setter function we should use it, for clarity of the code. Since this setter function is used everywhere, it should emit an event (otherwise we would not be able to reconstruct the state from events). So I guess it's option 1, not because it emits 2 events and we want to emit 2 events, but because it's the most convenient/clean in the code while still allowing to reconstruct the state

Reply:
Ive removed `PendingRootSet` emission when we emit `RootSet`